rust/cas_client/src/caching_client.rs

```diff
@@ -77,35 +72,34 @@ impl<T: Client + Debug + Sync + Send> Client for CachingClient<T> {
         hash: &MerkleHash,
         data: Vec<u8>,
         chunk_boundaries: Vec<u64>,
-    ) -> Result<(), CasClientError> {
+    ) -> Result<()> {
         // puts write through
         debug!(
             "CachingClient put to {}/{} of length {} bytes",
             prefix,
             hash,
             data.len()
         );
-        self.client.put(prefix, hash, data, chunk_boundaries).await
+        Ok(self
+            .client
+            .put(prefix, hash, data, chunk_boundaries)
+            .await?)
     }
 
-    async fn flush(&self) -> Result<(), CasClientError> {
+    async fn flush(&self) -> Result<()> {
         // forward flush to the underlying client
-        self.client.flush().await
+        Ok(self.client.flush().await?)
     }
 
-    async fn get(&self, prefix: &str, hash: &MerkleHash) -> Result<Vec<u8>, CasClientError> {
+    async fn get(&self, prefix: &str, hash: &MerkleHash) -> Result<Vec<u8>> {
         // get the length, reduce to range read of the entire length.
         debug!("CachingClient Get of {}/{}", prefix, hash);
-        let xorb_size = match self.get_length(prefix, hash).await {
-            Err(e) => {
-                debug!("CachingClient Get: get_length reported error : {e:?}");
-                return Err(e);
-            }
-            Ok(v) => {
-                debug!("CachingClient Get: get_length call succeeded with value {v}.");
-                v
-            }
-        };
+        let xorb_size = self.get_length(prefix, hash).await.map_err(|e| {
+            debug!("CachingClient Get: get_length reported error : {e:?}");
+            e
+        })?;
```

Comment:
for these map_err just used for logging we should use the ErrorPrinter trait, but may have to pull it out of gitxetcore into somewhere more usable.

https://github.com/xetdata/xet-core/blob/main/rust/gitxetcore/src/log.rs#L22

Reply:
I'm open to that; I think we could integrate it into the new xet_error class?

Reply:
we could do that, the way the ErrorPrinter trait works is it's already implemented for all Result<T, E> types, so I we don't need it for the special error type, but that would be a good area od the code to put it if it's in its own crate.